ui/main/src/app/modules/card/components/card-body/card-body.component.ts

```diff
@@ -258,9 +276,19 @@ export class CardBodyComponent implements OnChanges, OnInit, OnDestroy {
                 if (this.card.id !== this.lastCardSetToReadButNotYetOnFeed.id)
                     this.updateLastReadCardStatusOnFeedIfNeeded();
             }
-            this.lastCardSetToReadButNotYetOnFeed = this.card;
             CardService.postUserCardRead(this.card.uid).subscribe();
+            
         } else this.updateLastReadCardStatusOnFeedIfNeeded();
+
+        if (this.childCards) {
+            this.childCards.forEach(child => {
+                if (child.actions?.includes(CardAction.PROPAGATE_READ_ACK_TO_PARENT_CARD) && !child.hasBeenRead) {
+                    CardService.postUserCardRead(child.uid).subscribe();
+                }
+            })
+        }
+        this.lastCardSetToReadButNotYetOnFeed = this.card;
+
     }
```

Comment:
this.lastCardSetToReadButNotYetOnFeed = this.card shall only be done if read status change

So keep the old  this.lastCardSetToReadButNotYetOnFeed = this.card; 
and add a new after CardService.postUserCardRead(child.uid).subscribe();